Billing.API/Controllers/InvoiceController.cs

```diff
@@ -14,7 +14,7 @@ namespace Billing.API.Controllers
     [ApiController]
     public class InvoiceController : ControllerBase
     {
-        private static readonly string _invoiceFilenameRegex = @"^invoice_\d{4}-\d{2}-\d{2}_(\d+)\.pdf$";
+        private const string INVOICE_FILENAME_REGEX = @"^invoice_\d{4}-\d{2}-\d{2}_(\d+)\.pdf$";
```

Comment:
Since you are changing this code, what do you think about parsing regex only one time?

```suggestion
        private static readonly Regex INVOICE_FILENAME_REGEX = new Regex(@"^invoice_\d{4}-\d{2}-\d{2}_(\d+)\.pdf$");
```